qiskit/circuit/library/generalized_gates/permutation.py

```diff
@@ -88,3 +92,112 @@ def __init__(
 
         all_qubits = self.qubits
         self.append(circuit.to_gate(), all_qubits)
+
+
+class PermutationGate(Gate):
```

Comment:
Do you want to use this inside the `Permutation` class's definition. For example:

```python3
class Permutation(QuantumCircuit):

    def __init__(
        self,
        num_qubits: int,
        pattern: Optional[List[int]] = None,
        seed: Optional[int] = None,
    ) -> None:
        if pattern is not None:
            if sorted(pattern) != list(range(num_qubits)):
                raise CircuitError(
                    "Permutation pattern must be some ordering of 0..num_qubits-1 in a list."
                )
            pattern = np.array(pattern)
        else:
            rng = np.random.default_rng(seed)
            pattern = np.arange(num_qubits)
            rng.shuffle(pattern)
        circuit = QuantumCircuit(num_qubits, name=name)
        super().__init__(num_qubits, name=name)
        circuit.append(PermutationGate(pattern), self.qubits)
```

Or do you think the difference in behavior is too much?

Reply:
This is an interesting suggestion. It would probably work, but (as the PermutationGate has no definition) it could also in principle break something (like it happened for Cliffords). I did check and added tests for methods like `inverse`, `reversed_ops`, etc. there might be other circuit's methods that I have not fully considered. So maybe for now let's err on the safe side and keep the older implementation as is?